### ISSUE TITLE
Possible Bug: After PTRACE_INTERRUPT, `in_injectible_signal_stop` should be false

### ISSUE THREAD

**Issue by sidkshatriya** [rr-debugger/rr#3680]
I'm not sure about this one. However the code looked a little suspicious, so flagging this:

In `Task::did_waitpid()` we create a fake TIME_SLICE_SIGNAL when we see a SIGTRAP as a result of a PTRACE_INTERRUPT.

https://github.com/rr-debugger/rr/blob/d8ba5ac102ed4145b7de2c443a117bcfaf4969c4/src/Task.cc#L2190-L2197

Now the code thinks this was TIME_SLICE_SIGNAL and hence potentially a signal stop. 

In the same function `did_waitpid`, later we have:

https://github.com/rr-debugger/rr/blob/d8ba5ac102ed4145b7de2c443a117bcfaf4969c4/src/Task.cc#L2271-L2273

So after a PTRACE_INTERRUPT, we're really in non-injectible SIGTRAP but rr will set `in_injectible_signal_stop` to `true` when it should set it to `false` ??

**Comment by rocallahan** [comment]
I think you're right.

**Comment by rocallahan** [comment]
This test program suggests that ` PTRACE_INTERRUPT` `SIGTRAP` is indeed non-injectible:
```
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>
#include <sys/ptrace.h>
#include <sys/wait.h>

static void test_assert(int condition) {
  if (!condition) {
    abort();
  }
}

static int parent_to_child_fds[2];
static int child_to_parent_fds[2];

int main(void) {
  pid_t child;
  char ch;
  int status;

  test_assert(0 == pipe(parent_to_child_fds));
  test_assert(0 == pipe(child_to_parent_fds));

  if (0 == (child = fork())) {
    test_assert(1 == read(parent_to_child_fds[0], &ch, 1));
    test_assert(1 == write(child_to_parent_fds[1], "x", 1));
    
    volatile int ccc = 0;
    while (!ccc) {
    }
    
    return 77;
  }

  test_assert(0 ==
              ptrace(PTRACE_SEIZE, child, NULL, (void*)PTRACE_O_TRACESYSGOOD));
  /* Make sure child is still running */
  test_assert(1 == write(parent_to_child_fds[1], "p", 1));
  test_assert(1 == read(child_to_parent_fds[0], &ch, 1));

  sleep(5);
  test_assert(0 == ptrace(PTRACE_INTERRUPT, child, NULL, 0));
  test_assert(child == waitpid(child, &status, 0));
  printf("Got status %x\n", status);
  test_assert(0 == ptrace(PTRACE_CONT, child, NULL, (void*)SIGTERM));
  sleep(60);
  return 0;
}
```

**Comment by sidkshatriya** [comment]
Thanks ! 

Strangely now the `ptrace_seize` test is no longer succeeding (I notice you have made changes to this to reflect the test case above).

I tried on a fedora tigerlake and a ryzen.

**Comment by sidkshatriya** [comment]
The new `ptrace_seize` basically times out on fedora 6.6.11-200.

---
When I had a quick look, the problem seems to be due to 
```
    volatile int ccc = 0;
    while (!ccc) {
    }
```

Essentially `PTRACE_INTERRUPT` is not able to interrupt this pure user space loop. Is that expected ? 

One way to make this effect more apparent is to change the sleep(60) to something like sleep(1).

https://github.com/rr-debugger/rr/blob/fabdbc9c0373fca3f6d2daba3835e42045b34b6a/src/test/ptrace_seize.c#L49

**Comment by sidkshatriya** [comment]
Ok, there is a bug in the code. There is a `millisecond` to `second` confusion.

https://github.com/rr-debugger/rr/blob/fabdbc9c0373fca3f6d2daba3835e42045b34b6a/src/test/ptrace_seize.c#L58

This should be `sleep(1)` and not `sleep(1000)`.

**Now the test works when _not_ running in rr but can assert when running under rr.**

```
[FATAL src/Task.cc:1179:extra_regs() errno: ESRCH] 
 (task 43933 (rec:43933) at time 251)
 -> Assertion `false' failed to hold. Can't find task for infallible extra_regs
```

**Comment by rocallahan** [comment]
I checked some changes I made while doing experiments that weren't supposed to be committed.

**Comment by rocallahan** [comment]
Fixed in 6523fe58c1925ee8b1c16e5363a2d6f46bdb53a1
